controllers/volumereplicationgroup_controller.go

```diff
@@ -861,13 +874,25 @@ func (v *VRGInstance) reconcileVRsAsPrimary() bool {
 		if _, err := v.processVRAsPrimary(pvcNamespacedName, log); err != nil {
 			log.Info("Requeuing due to failure in getting or creating VolumeReplication resource for PersistentVolumeClaim",
 				"errorValue", err)
+		}
 
-			requeue = true
+		// Protect the PVC's PV object stored in etcd by uploading it to S3
+		// store(s).  Note that the VRG is responsible only to protect the PV
+		// object of each PVC of the subscription.  However, the PVC object
+		// itself is assumed to be protected along with other k8s objects in the
+		// subscription, such as, the deployment, pods, services, etc., by an
+		// entity external to the VRG a la IaC.
+		if err := v.uploadPVToS3Stores(pvc, log); err != nil {
+			log.Info("Requeuing due to failure to upload PV object to S3 store(s)",
+				"errorValue", err)
 
+			requeue = true
+			// TODO: should we use requeueAfter some time duration to retry
+			// uploading to S3 store again, instead of requeing immediately?
 			continue
 		}
 
-		log.Info("Successfully processed VolumeReplication for PersistentVolumeClaim")
+		log.Info("Successfully protected the PVC as primary")
```

Comment:
We setup logging with values state (key) and primary|secondary (value) and also filalize (key) with true (value) in processVRGActions, which helps us avoid primary|secondary words in the log message. I would hence stick to the older message here.